ast/edit/task_editor.go

```diff
@@ -0,0 +1,138 @@
+package edit
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/codes"
+)
+
+// GetOption finds and returns the AST node for the option value.
+func GetOption(pkg *ast.Package, name string) (ast.Node, error) {
+	for _, file := range pkg.Files {
+		for _, st := range file.Body {
+			val, ok := st.(*ast.OptionStatement)
+			// wrong statement type; skip this statement
+			if !ok {
+				continue
+			}
+			assign := val.Assignment
+			va, ok := assign.(*ast.VariableAssignment)
+			if va.ID.Name == name {
+				if ok {
+					return val, nil
+				}
+			}
+		}
+	}
+	return nil, &flux.Error{
+		Code: codes.Internal,
+		Msg:  "Option not found",
+	}
+}
+
+// SetOption replaces an existing option definition with the provided option node or adds
+// the option if it doesn't exist. The pkg AST is mutated in place.
+func SetOption(pkg *ast.Package, name string, id string, opt ast.Node) error {
+	for _, file := range pkg.Files {
+		// check for the correct file
+		if file.Name == name {
+			for i, st := range file.Body {
+				val, ok := st.(*ast.OptionStatement)
+				// wrong statement type; skip this statement
+				if !ok {
+					continue
+				}
+				assign := val.Assignment
+				va, ok := assign.(*ast.VariableAssignment)
+				if ok {
+					if va.ID.Name == id {
+						if optSt, ok := opt.(*ast.OptionStatement); ok {
+							file.Body[i] = optSt
+							return nil
+						}
+					}
+				}
+
+			}
+			// found file, but did not find option; add option to body
+			file.Body = append(file.Body, &ast.OptionStatement{
+				BaseNode:   ast.BaseNode{},
+				Assignment: nil,
+			})
+		}
+	}
+	return &flux.Error{
+		Code: codes.Internal,
+		Msg:  fmt.Sprintf("File with name %v, not found", name),
+	}
+}
+
+// DeleteOption removes an option if it exists. The pkg AST is mutated in place.
+func DeleteOption(pkg *ast.Package, name string) error {
+	for _, file := range pkg.Files {
+		for i, st := range file.Body {
+			val, ok := st.(*ast.OptionStatement)
+			if !ok {
+				continue
+			}
+			assign := val.Assignment
+			va, ok := assign.(*ast.VariableAssignment)
+			if va.ID.Name == name {
+				if ok {
+					file.Body = append(file.Body[:i], file.Body[i+1:]...)
+					return nil
+				}
+			}
+		}
+	}
+	return &flux.Error{
+		Code: codes.Internal,
+		Msg:  fmt.Sprintf("Could not delete %v, Option not found.", name),
+	}
+}
+
+// GetProperty finds and returns the AST node for the property value.
+func GetProperty(obj *ast.ObjectExpression, key string) (ast.Node, error) {
```

Comment:
Should this return an `*ast.Property` since we know it will always be that type?